File: TaskManagement.API/DTOs/TodoTaskDto.cs
using TaskManagement.API.Enums;

namespace TaskManagement.API.DTOs;

public class TodoTaskDto
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DateOnly DueDate { get; set; }
    public TaskCompletionStatus Status { get; set;}
    public string Remarks { get; set; }
    public UserMinDto User { get; set; }
    public long CreatedOn { get; set; }
    public long LastUpdatedOn { get; set; }
}

File: TaskManagement.Web/src/app/dtos/TodoTaskDto.ts
import { TaskCompletionStatus } from "../enums/TaskCompletionStatus";
import { UserMinDto } from "./UserDto";

export interface TodoTaskDto {
    id?: string;
    title: string;
    description: string;
    dueDate: string; // HTML date inputs and .NET DateOnly use "YYYY-MM-DD"
    status: TaskCompletionStatus;
    remarks: string;
    user?: UserMinDto;
    createdOn?: number;
    lastUpdatedOn?: number;
}